Examples/Resources/Extractors/SAT-features-competition2012_revised_without_SatELite_sparkle/sparkle_extractor_wrapper.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+"""Template for users to create Feature Extractor wrappers."""
+
+import sys
+import os
+import time
+import random
+import argparse
+import subprocess
+from pathlib import Path
+
+global sparkle_special_string
+sparkle_special_string = r'__@@SPARKLE@@__'
+
+def get_last_level_directory_name(filepath):
+    return os.path.basename(filepath.rstrip('/'))
+
+def get_time_pid_random_string():
+    return f"{time.strftime('%Y%m%d-%H%M%S')}_{os.getpid()}_{random.randint(1000000, 9999999)}"
+
+parser = argparse.ArgumentParser(description="Process some integers.")
+parser.add_argument('-extractor_dir', type=str, help='Path to the extractor directory')
+parser.add_argument('-instance_file', type=str, help='Path to the instance file')
+parser.add_argument('-output_file', type=str, help='Path to the output file')
+args = parser.parse_args()
+
+extractor_dir = args.extractor_dir
+instance_path = args.instance_file
+output_file = args.output_file
+
+executable_name = "features"
+
+raw_result_file_name = f"{extractor_dir}{executable_name}_" \
+                      f"{get_last_level_directory_name(instance_path)}_" \
+                      f"{get_time_pid_random_string()}.rawres"
+tmp_output = f"TMP/{raw_result_file_name}"
+
+command_line = [f"{extractor_dir}/{executable_name}", instance_path, tmp_output]
+
+with Path(raw_result_file_name).open("w+") as raw_result_file:
+    subprocess.run(command_line, stdout=raw_result_file)
+
+#Processes raw result file and writes to the final result file
+with open(raw_result_file_name, 'r') as fin, open(output_file, 'w') as fout:
+        while True:
+            # Reads every line, strips whitespace and returns on empty line
+            myline = fin.readline().strip()
+            if not myline:
+                break
+            
+            # Splits the line into single words and skips empty lines & lines starting with 'c'
+            mylist = myline.split()
+            if len(mylist) == 0 or mylist[0] == 'c':
+                continue
+            
+            # Splits the line by commas, writes each part to the output file with additional information
+            mylist_comma = myline.split(',')
+            for item in mylist_comma:
+                fout.write(f',{item}{sparkle_special_string}{get_last_level_directory_name(extractor_dir)}')
+            fout.write('\n')
+
+            myline2 = fin.readline().strip()
+            fout.write(f'{instance_path},{myline2}\n')
+
+# Deletes temporary files
+Path(tmp_output).unlink(missing_ok=True)
```

Comment:
Make sure the variables are already paths at definition so you don't have to do Path() here. Also why are these files being deleted?

Reply:
Alright check if this still makes sense

Reply:
Only "raw_result_file_name.unlink(missing_ok=True)" seems to be necessary. It references the file "_Extractors/SAT-features-competition2012_revised_without_SatELite_sparklefeatures_sparkle_test_instance.cnf_20240612-105923_1773317238.rawres_"
The path tmp_output should probably refrence the .rawres file in the TMP folder, however there is a slight difference in the formatting of the path and therefor tmp_output points to nothing. However the file it should point to is removed anyway.

Reply:
I think I know what it is.

You asked me to remove the sparkle_special_string but it turns out that the special_string is used as a seperator in the fdch.get_extractor_path_from_feature function.

For now I'll implement the special string again, if you want we can replace it with something else.